cxx/transition_true_value.hh

```diff
@@ -0,0 +1,105 @@
+// Copyright 2024
+// See LICENSE.txt
+
+#pragma once
+
+#include <algorithm>
+#include <random>
+#include <string>
+#include <unordered_map>
+
+#include "emissions/base.hh"
+#include "noisy_relation.hh"
+#include "relation.hh"
+
+typedef int T_item;
+typedef std::vector<T_item> T_items;
+
+// Updates the estimate of a latent value conditioned on noisy observations.
+template <typename ValueType>
+void transition_true_value(
```

Comment:
Can this be broken up into two methods: Computing the logp associated with the different possibilities, and the actual selection / transitioning? The thought is we could test correctness of the first part (that we are computing the correct incremental logp for some synthetic example).

Reply:
That's my high level sense too re. Model 6/transition_corrupted, though I don't have all the details worked out.

I wasn't sure of the best way to break it up -- let me know what you think of this.